### PR TITLE
Fix bug 1243126. Duplicated results should be grouped by the source i…

### PR DESCRIPTION
**Pull request by jotes**
hi @mathjazz, could you review?

### CODE REVIEW

**Comment by mathjazz** [comment]
Nice work!

You shouild also update Machinery tab count, which should count a group of 3 duplicates as 1 instead of 3.

There's also no need to display % for all sources, because it's the same for all duplicates. That way you can list all sources in one line to reduce some whitespace.

So what currently looks like this:
```
100% • #14 Translation memory
100% Microsoft
100% Mozilla
```

Should look like this:
```
100% • #14 Translation memory • Mozilla • Microsoft
```

**Comment by jotes** [comment]
@mathjazz could you look again?

File: pontoon/base/static/js/main.js [hunk 1]
```diff
@@ -126,24 +126,35 @@ var Pontoon = (function (my) {
...
-              'title="' + data.title + '">' + data.source + '</a>' +
-          '</header>' +
-          '<p class="original">' + self.doNotRender(data.original || '') + '</p>' +
-          '<p class="translation" dir="auto" lang="' + self.locale.code + '">' +
-            self.doNotRender(data.translation) +
-          '</p>' +
-        '</li>');
-
+        var title = loader !== 'search' ? ' title="Copy Into Translation (Tab)"' : '',
+            sources = sourcesMap[data.original+data.translation];
```
Comment (mathjazz) [inline pontoon/base/static/js/main.js]: Add space before and after `+`.

File: pontoon/base/static/js/main.js [hunk 2]
```diff
@@ -126,24 +126,35 @@ var Pontoon = (function (my) {
...
+              '<a class="translation-source" href="' + data.url + '" target="_blank"' +
+                'title="' + data.title + '">' + data.source + '</a>' +
+            '</header>' +
+            '<p class="original">' + self.doNotRender(data.original || '') + '</p>' +
+            '<p class="translation" dir="auto" lang="' + self.locale.code + '">' +
+              self.doNotRender(data.translation) +
+            '</p>' +
+          '</li>');
+          ul.append(li);
+          sourcesMap[data.original+data.translation] = li.children('header');
```
Comment (mathjazz) [inline pontoon/base/static/js/main.js]: Add space before and after `+`.

**Comment by mathjazz** [comment]
Nice job!

I'd like to ask you for additional frontend improvement. It's not clear from the design that the number of TM matches actually belongs to TM only. How about we use exponent for this purpose?

```
(venv)Snowball:pontoon mathjazz$ git diff
diff --git a/pontoon/base/static/css/style.css b/pontoon/base/static/css/style.css
index cfa8eb2..2ade30a 100755
--- a/pontoon/base/static/css/style.css
+++ b/pontoon/base/static/css/style.css
@@ -1179,6 +1179,14 @@ body > .container {
   padding-right: 3px;
 }
 
+#helpers > section.machinery ul li > header .stress {
+  padding-right: 0;
+}
+
+#helpers > section.machinery ul li > header sup {
+  color: #7BC876;
+}
+
 #helpers > section ul li > p {
   min-height: 22px;
   text-align: left;
diff --git a/pontoon/base/static/js/main.js b/pontoon/base/static/js/main.js
index 7ae15a4..601323c 100755
--- a/pontoon/base/static/js/main.js
+++ b/pontoon/base/static/js/main.js
@@ -141,10 +141,12 @@ var Pontoon = (function (my) {
           var li = $('<li' + title + '>' +
             '<header>' +
               '<span class="stress">' + (data.quality || '') +
-                (data.count ? ' &bull; <span>#</span>' + data.count : '') +
               '</span> &bull; ' +
               '<a class="translation-source" href="' + data.url + '" target="_blank"' +
-                'title="' + data.title + '">' + data.source + '</a>' +
+                'title="' + data.title + '">' +
+                data.source +
+                (data.count ? '<sup>' + data.count + '</sup> ' : '') +
+              '</a>' +
             '</header>' +
             '<p class="original">' + self.doNotRender(data.original || '') + '</p>' +
             '<p class="translation" dir="auto" lang="' + self.locale.code + '">' +
```

**Comment by jotes** [comment]
Hi, could you look again?

File: pontoon/base/static/js/main.js [hunk 3]
```diff
@@ -126,24 +126,40 @@ var Pontoon = (function (my) {
...
+            ' &bull; <a class="translation-source" href="' + data.url + '" target="_blank" title="' + data.title + '">' +
+              '<span>' + data.source + '</span>' +
+              (data.count ? '<sup>' + data.count  + '</sup>' : '') +
+            '</a>'
+          );
+        } else {
+          var li = $('<li' + title + '>' +
+            '<header>' +
+              '<span class="stress">' + (data.quality || '') +
+                (data.count ? ' &bull; <span>#</span>' + data.count : '') +
```
Comment (mathjazz) [inline pontoon/base/static/js/main.js]: Remove this line.

**Comment by mathjazz** [comment]
Almost there. ;) Count appear two times now.

**Comment by jotes** [comment]
I had this fixed locally, but forgot :D I'm sorry for the trouble. Could you check again?

**Comment by mathjazz** [comment]
A thing of beauty! ;) r+

**Pull request merged**